### PR TITLE
enhancement(codecs): Serialize metrics in `RawMessageSerializer`

### PR DESCRIPTION
**Pull request by pablosichert**
This came up when integrating the new `Encoder` with the `console` sink in #12181.

### CODE REVIEW

File: lib/codecs/src/encoding/format/raw_message.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ impl Encoder<Event> for RawMessageSerializer {
                 .get_by_meaning(message_key)
                 .or_else(|| log.get(message_key)) // backward compatibility
                 .map(|value| value.coerce_to_bytes()),
-            Event::Metric(_) => None,
+            Event::Metric(metric) => Some(metric.to_string().into()),
```
Comment (jszwedko) [inline lib/codecs/src/encoding/format/raw_message.rs:51]: Is this prometheus exposition format? Or something else?

**Review by jszwedko — APPROVED** [review]

**Comment by jszwedko** [comment]
Also, is `RawMessageSerializer` the analogue of `BytesDeserializer`? I wonder if we should unify the naming to make that relationship clearer.

File: lib/codecs/src/encoding/format/raw_message.rs (hunk 1 above)
Comment (pablosichert) [reply lib/codecs/src/encoding/format/raw_message.rs:51]: Something similar I suppose:

https://github.com/vectordotdev/vector/blob/1c87d918b53b04c88274e12027519a1c27a55741/lib/vector-core/src/event/metric/mod.rs#L361-L382

**Comment by pablosichert** [comment]
> Also, is `RawMessageSerializer` the analogue of `BytesDeserializer`? I wonder if we should unify the naming to make that relationship clearer.

I can see the argument for that. The `BytesDeserializer` has no concept of a message key, though, so it wouldn't be exactly symmetrical

**Comment by jszwedko** [comment]
> BytesDeserializer

Hmm, doesn't it? It seems to create a log event with the raw payload as `message` (or whatever the user has configured as the log schema message field):

https://github.com/vectordotdev/vector/blob/b421e024af6dc6bb91a85a5afd5483347707b1e6/lib/codecs/src/decoding/format/bytes.rs#L68-L72

**Comment by pablosichert** [comment]
@jszwedko indeed... Must have confused something there 🤔

Opened #12529 to keep track of it

**Review by lukesteensen — COMMENTED** [review]
I would actually lean towards not supporting metrics with a `RawMessageSerializer` given that metrics definitely do not have raw messages. It feels like this is a good example of why we need an equivalent of #12049 but for sinks.

**Comment by jszwedko** [comment]
> I would actually lean towards not supporting metrics with a `RawMessageSerializer` given that metrics definitely do not have raw messages. It feels like this is a good example of why we need an equivalent of #12049 but for sinks.

Yeah, that's a good point. We could limit the input types based on the supplied codec. I like this too.

**Pull request closed without merging**